### PR TITLE
177868 - Add event queue service

### PR DESCRIPTION
**Pull request by JasmeetKaur-Defra**

### CODE REVIEW

File: EST.MIT.InvoiceImporter.Function/Configuration/ServiceConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using Azure.Storage.Queues;
+using EST.MIT.InvoiceImporter.Function.Interfaces;
+using EST.MIT.InvoiceImporter.Function.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Win32;
+
+namespace EST.MIT.InvoiceImporter.Function.Configuration
+{
```
Comment (gldefra) [inline EST.MIT.InvoiceImporter.Function/Configuration/ServiceConfiguration.cs:8]: Please update to a file scoped namespace for consistency.

File: EST.MIT.InvoiceImporter.Function/Functions/Importer.cs [hunk 2]
```diff
@@ -28,6 +30,8 @@ public Importer(IBlobService blobService, IConfiguration configuration, IAzureBl
         IBinder blobBinder,
         ILogger log)
     {
+        //event queue test
```
Comment (gldefra) [inline EST.MIT.InvoiceImporter.Function/Functions/Importer.cs]: Is this intended ?

File: EST.MIT.InvoiceImporter.Function/Models/EventAction.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+public class EventAction
+{
+    [JsonPropertyName("type")]
+    public string Type { get; init; } = default!;
+    [JsonPropertyName("message")]
+    public string Message { get; init; } = default!;
+    [JsonPropertyName("timestamp")]
+    public DateTime Timestamp { get; init; } = default!;
+    [JsonPropertyName("data")]
+    public string Data { get; init; } = default!;
```
Comment (gldefra) [inline EST.MIT.InvoiceImporter.Function/Models/EventAction.cs:14]: Should this be an object?

File: EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+    private readonly QueueClient _queueClient;
+
+    public EventQueueService(QueueClient queueClient)
+    {
+        _queueClient = queueClient;
+    }
+
+    public async Task CreateMessage(string id, string status, string action, string message, Invoice? invoice = null)
+    {
```
Comment (gldefra) [inline EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs:30]: Does CreateMessage need any null checks?

File: EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                    Type = action,
+                    Message = message,
+                    Timestamp = DateTime.UtcNow,
+                    Data = invoice == null ? null : JsonSerializer.Serialize(invoice)
+                }
+            }
+        };
+
+        var bytes = Encoding.UTF8.GetBytes(JsonSerializer.Serialize(eventRequest));
+        await _queueClient.SendMessageAsync(Convert.ToBase64String(bytes));
```
Comment (gldefra) [inline EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs]: Does this need to be wrapped with an exception handler?

**Review by gldefra — APPROVED** [review]

File: EST.MIT.InvoiceImporter.Function/Configuration/ServiceConfiguration.cs (hunk 1 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.InvoiceImporter.Function/Configuration/ServiceConfiguration.cs:8]: Resolved

File: EST.MIT.InvoiceImporter.Function/Functions/Importer.cs (hunk 2 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.InvoiceImporter.Function/Functions/Importer.cs]: Removed

File: EST.MIT.InvoiceImporter.Function/Models/EventAction.cs (hunk 3 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.InvoiceImporter.Function/Models/EventAction.cs:14]: Can you please elaborate this?

File: EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs (hunk 4 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs:30]: Do you mean the parameter needs to be checked against null?

File: EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs (hunk 5 above)
Comment (JasmeetKaur-Defra) [reply EST.MIT.InvoiceImporter.Function/Services/EventQueueService.cs]: Resolved

**Pull request closed without merging**